lib/classes/class-db.php

```diff
@@ -0,0 +1,886 @@
+          `self_link` text NOT NULL,
+          PRIMARY KEY (`id`),
+          KEY post_id (post_id),
+          UNIQUE KEY post_id_size (post_id, size_name)
+        ) $charset_collate;
+
+        CREATE TABLE $this->file_meta (
+          `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT,
+          `post_id` bigint(20) unsigned NULL DEFAULT NULL,
+          `meta_key` varchar(255) NOT NULL,
+          `meta_value` longtext NOT NULL,
+          PRIMARY KEY (`id`),
+          KEY post_id (post_id),
+          UNIQUE KEY post_id_key (post_id, meta_key)
+        ) $charset_collate;";
+
+        require_once(ABSPATH . 'wp-admin/includes/upgrade.php');
+        dbDelta($sql);
+
+        update_site_option('sm_db_version', self::DB_VERSION);
+      }
+
+      /**
+       * Remove custom DB table on plugin uninstall
+       */
+      public function clear_db() {
+        $sql = "DROP TABLE IF EXISTS $this->files, $this->file_sizes, $this->file_meta;";
+        $wpdb->query($sql);
+      }
+
+      /**
+       * Get file link
+       * 
+       * @param string $name
+       * @return string
+       */
+      public function get_file_link($name) {
+        return trailingslashit($this->bucket_link) . $name;
+      }
+
+      /**
+       * Get file row ID by post ID
+       * 
+       * @param int $post_id
+       * @return int | null
+       */
+      public function get_file_id($post_id) {
+        $sql = "SELECT id FROM $this->files WHERE post_id = %d";
+        $id = $this->wpdb->get_var( $this->wpdb->prepare($sql, $post_id) );
+
+        return $id ? (int) $id : null;
+      }
+
+      /**
+       * Get file size row ID by post ID and size name
+       * 
+       * @param int $post_id
+       * @param string $size
+       * @return int | null
+       */
+      public function get_file_size_id($post_id, $size) {
+        $sql = "SELECT id FROM $this->file_sizes WHERE post_id = %d AND size_name = %s";
+        $id = $this->wpdb->get_var( $this->wpdb->prepare($sql, $post_id, $size) );
+
+        return $id ? (int) $id : null;
+      }
+
+      /**
+       * Get file meta row ID by post ID and meta key
+       * 
+       * @param int $post_id
+       * @param string $key
+       * @return int | null
+       */
+      public function get_file_meta_id($post_id, $key) {
+        $key = sanitize_key($key);
+
+        $sql = "SELECT id FROM $this->file_meta WHERE post_id = %d AND meta_key = %s";
+        $id = $this->wpdb->get_var( $this->wpdb->prepare($sql, $post_id, $key) );
+
+        return $id ? (int) $id : null;
+      }
+
+      /**
+       * Add file data to DB
+       * 
+       * @param array $cloud_meta   - generated WP postmeta
+       * @param array $media        - GCS media object
+       * @param string $image_size  - image size name
+       * @param array $img          - image (or image size) data for upload
+       * @param string $bucketLink  - bucket link
+       * @param int $attachment_id  - attachment ID
```

Comment:
@balexey88 this line can be removed

Reply:
Done https://github.com/udx/wp-stateless/commit/e93b7550742f3e31acc6e30449340e30b70588b0